hack/generator/pkg/astmodel/arm_type.go

```diff
@@ -26,7 +26,7 @@ func MakeArmType(object ObjectType) ArmType {
 }
 
 // RequiredImports returns a list of packages required by this type
-func (at ArmType) RequiredImports() []PackageReference {
+func (at ArmType) RequiredImports() *PackageImportSet {
```

Comment:
As mentioned above in long comment, I think this is conflating two things (Import vs Reference)

Won't repeat comment on each location though as there are a number of occurrences.